### PR TITLE
refactor: split declaration by event type

### PR DESCRIPTION
**Pull request by pevisscher**

### CODE REVIEW

**Comment by pevisscher** [comment]
Current dependencies on/for this PR:
* main
  * **PR #11** <a href="https://app.graphite.dev/github/pr/skyleague/event-horizon/11" target="_blank">Graphite</a> 
    * **PR #12** <a href="https://app.graphite.dev/github/pr/skyleague/event-horizon/12" target="_blank">Graphite</a> 
      * **PR #14** <a href="https://app.graphite.dev/github/pr/skyleague/event-horizon/14" target="_blank">Graphite</a> 
        * **PR #13** <a href="https://app.graphite.dev/github/pr/skyleague/event-horizon/13" target="_blank">Graphite</a> 
          * **PR #15** <a href="https://app.graphite.dev/github/pr/skyleague/event-horizon/15" target="_blank">Graphite</a> 
            * **PR #16** <a href="https://app.graphite.dev/github/pr/skyleague/event-horizon/16" target="_blank">Graphite</a> 
              * **PR #17** <a href="https://app.graphite.dev/github/pr/skyleague/event-horizon/17" target="_blank">Graphite</a> 
                * **PR #18** <a href="https://app.graphite.dev/github/pr/skyleague/event-horizon/18" target="_blank">Graphite</a> 
                  * **PR #19** <a href="https://app.graphite.dev/github/pr/skyleague/event-horizon/19" target="_blank">Graphite</a> 
                    * **PR #20** <a href="https://app.graphite.dev/github/pr/skyleague/event-horizon/20" target="_blank">Graphite</a> 
                      * **PR #21** <a href="https://app.graphite.dev/github/pr/skyleague/event-horizon/21" target="_blank">Graphite</a> 
                        * **PR #22** <a href="https://app.graphite.dev/github/pr/skyleague/event-horizon/22" target="_blank">Graphite</a>  👈
                          * **PR #23** <a href="https://app.graphite.dev/github/pr/skyleague/event-horizon/23" target="_blank">Graphite</a> 
                            * **PR #24** <a href="https://app.graphite.dev/github/pr/skyleague/event-horizon/24" target="_blank">Graphite</a> 

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/skyleague/event-horizon/22?utm_source=stack-comment).

File: src/lib/event.ts
Comment (MickVanDuijn) [inline src/lib/event.ts:160]: I'm afraid these names are too ambiguous, and will clash with application code (same for `http`, `sns`, etc). what about `kinesisHandler` or `kinesisEventHandler`?

File: src/lib/event.ts
Comment (MickVanDuijn) [inline src/lib/event.ts:87]: in light of the other comment, maybe we should rename this to `eventHandler`?

**Review by MickVanDuijn — CHANGES_REQUESTED** [review]
some discussion points for naming

**Review by MickVanDuijn — APPROVED** [review]

**Comment by pevisscher** [comment]
**Graphite Merge Job**

Current status: ⏳ Queued to merge

This pull request is currently queued to be merged as part of a stack.

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/skyleague/event-horizon/22?utm_source=stack-comment).


Job Reference: 74QS0D26BjCoc1fjew09

**Comment by pevisscher** [comment]
**Graphite Merge Job**

Current status: ⏳ Queued to merge

This pull request is currently queued to be merged as part of a stack.

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/skyleague/event-horizon/22?utm_source=stack-comment).


Job Reference: MNQ6fIHHP2CutIWmkx44

**Comment by pevisscher** [comment]
**Graphite Merge Job**

Current status: 🔄 Waiting for checks to pass

This pull request is currently waiting for CI and mergeability checks to pass before being merged.

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/skyleague/event-horizon/22?utm_source=stack-comment).


Job Reference: ZixEesAXj0mMWpqHV7EA

**Pull request merged**
